soc/arm/st_stm32/common/Kconfig.defconfig.series

```diff
@@ -25,8 +28,20 @@ endif # LOG_BACKEND_SWO
 
 # set the tick per sec as a divider of the LPTIM clock source
 config SYS_CLOCK_TICKS_PER_SEC
-	default 4096 if STM32_LPTIM_TIMER && STM32_LPTIM_CLOCK_LSE
-	default 4000 if STM32_LPTIM_TIMER && STM32_LPTIM_CLOCK_LSI
+	default 4096 if "$(dt_node_int_prop_int,$(DT_STM32_LPTIM_PATH),$(DT_ST_PRESCALER))" < 16
+	default 2048 if "$(dt_node_int_prop_int,$(DT_STM32_LPTIM_PATH),$(DT_ST_PRESCALER))" = 16
+	default 1024 if "$(dt_node_int_prop_int,$(DT_STM32_LPTIM_PATH),$(DT_ST_PRESCALER))" = 32
+	default  512 if "$(dt_node_int_prop_int,$(DT_STM32_LPTIM_PATH),$(DT_ST_PRESCALER))" = 64
+	default  256 if "$(dt_node_int_prop_int,$(DT_STM32_LPTIM_PATH),$(DT_ST_PRESCALER))" = 128
+	depends on STM32_LPTIM_TIMER && STM32_LPTIM_CLOCK_LSE
```

Comment:
If possible, `STM32_LPTIM_CLOCK_LSI` and `STM32_LPTIM_CLOCK_LSE` should be derived automatically from the kernel clock source of the LPTIM defined in the dts.
That way, we could really get rid of the old way of setting the source clock.
You'll probably need to set the value of `STM32_SRC_LSI` and `STM32_SRC_LSE` to be identical across all series (but they're just an arbitrary index, so that shouldn't be a problem).

Reply:
The Kconfig should get the STM32_LPTIM_CLOCK_LSE or STM32_LPTIM_CLOCK_LSI directly from the device Tree. However there is no $(dt_inst_clocks_cell_by_idx(0, 1, bus)  macro available to retrieve the <clocks> property from the deviceTree.